### ISSUE TITLE
Sparrow install script infinite loop

### ISSUE THREAD

**Issue by davenquinn** [EarthCubeGeochron/Sparrow#162]
This report was provided by email by Clinton Baker at NAU:

I'm following the docs here: https://sparrow-data.org/docs/getting-started.  The good news is that I'm not getting docker errors with this setup.  Or I haven't yet.

I'm running ubuntu 20.04 since it is easiest to get docker installed on and is probably what you developed this on.  However, the first step under the Bundled Version section fails when running:

```
curl -fsSL https://raw.githubusercontent.com/EarthCubeGeochron/Sparrow/HEAD/get-sparrow.sh | bash -s -
```

and I get this running as root or an unprivileged user:

```
To install to path elevated permissions are required!
If you choose to continue you will be prompted for your password.
Would you like to continue? (y/N)
No permissions granted.
aborting...
```

It looks like it is trying to prompt for a y/N response, but it just skips past it - it never stops to await input.

If I just download the get-sparrow.sh and run it, it stops to prompt me.  Answering y asks for a password (printed in plain text on the screen, which is an interesting choice).  As root, after typing a password, it seems to hang for a bit, then dumps out the same text above, but this time doesn't pause after asking y/N.  i.e.:

```
root@mygeeto:~# bash ./get-sparrow.sh
To install to path elevated permissions are required!
If you choose to continue you will be prompted for your password.
Would you like to continue? (y/N)
y
Please enter your password.
<root_password>
To install to path elevated permissions are required!
If you choose to continue you will be prompted for your password.
Would you like to continue? (y/N)
No permissions granted.
aborting...
```

As an unprivileged user, running the command in the instructions is the same as for root, but if I run the get-sparrow.sh script, I get slightly different result.  Note that is pauses both times asking for a password - the first time is a sudo prompt:

```
$ bash get-sparrow.sh
To install to path elevated permissions are required!
If you choose to continue you will be prompted for your password.
Would you like to continue? (y/N)
y
Please enter your password.
[sudo] password for cwb32:
To install to path elevated permissions are required!
If you choose to continue you will be prompted for your password.
Would you like to continue? (y/N)
y
Please enter your password.
<my password>
To install to path elevated permissions are required!
If you choose to continue you will be prompted for your password.
Would you like to continue? (y/N)
No permissions granted.
aborting...
```

I'm guessing some logic is bad in your install script.  The non-privileged user almost seems to work, but seems to loop back on itself.  If you take feature requests, it would be nice to let the software install like almost every other piece of software - allow the install to be performed as root (as root or via sudo).  Then if a non-privileged user needs to use it, it writes whatever user-specific bits its needs the first time the user runs the program (i.e. $HOME/.sparrow/whatever).  It would simplify your installation process greatly.

**Comment by davenquinn** [comment]
@Idzikowski-Casey  has been working on the install script over the last few days…I’m not sure if this pre- or postdates his changes. The reason for the work was something similar to what you have encountered. We will be sure to fix.

The reason we are elevating permissions inside the script is because we want to be able to request permissions and elevate only for the important action. We were modeling this off how the “home-brew” package manager seems to work. I agree it’s not the simplest when it fails.

We should also maintain instructions to install manually (first download, then sudo unzip to the right directory) alongside this attempted “easy" script.

**Comment by davenquinn** [comment]
We have likely fixed this in pull request #170
